x/bank/types/events.go

```diff
@@ -47,7 +47,16 @@ func NewCoinReceivedEvent(receiver sdk.AccAddress, amount sdk.Coins) sdk.Event {
 
 // NewCoinMintEvent construct a new coin minted sdk.Event
 // nolint: interfacer
-func NewCoinMintEvent(minter sdk.AccAddress, amount sdk.Coins) sdk.Event {
+func NewCoinMintEvent(minter sdk.AccAddress, amount sdk.Coins, receiver sdk.AccAddress) sdk.Event {
+	if len(receiver) > 0 {
+		return sdk.NewEvent(
+			EventTypeCoinMint,
+			sdk.NewAttribute(AttributeKeyMinter, minter.String()),
+			sdk.NewAttribute(sdk.AttributeKeyAmount, amount.String()),
+			sdk.NewAttribute(AttributeKeyReceiver, receiver.String()),
+		)
```

Comment:
do you think it make sense to have variable list of attributes for the same event type?

Reply:
why can the receiver be empty? a module has to mint coins somewhere, either to itself or to an address?

Reply:
Let's not diverge from the existing event definitions. For minting and sending let's just have two events and not add receiver to the mint event.

Reply:
@fdymylja  - receiver is empty to be compatible with the current behavior when minting to a module.
@aaronc - the idea was to not send and have a functionality and event to mint to an account.

Re event: maybe we can add a third argument in both cases? Will it break any client? Do we know how this events are parsed?

Reply:
> receiver is empty to be compatible with the current behavior when minting to a module.

What does this mean exactly? In what cases is it empty?

> the idea was to not send and have a functionality and event to mint to an account.

I don't follow this.

Reply:
I didn't catch this empty receiver part either. Maybe having a concrete example would help clarify.

Anyways, Rosetta depends on those events.

There's a mint event to identify a coinbase transaction.
And the mint event is associated with a coin_received event, this gives the outlook of an UTXO blockchain which is easier to understand from rosetta's perspective.

Reply:
Anyways, I don't think there should be differences at how MintCoinsToModule and MintCoinsToAddress look from outside, the only reason for which we distinguish them is for permissions I assume. I think they should look the same way.

Some module mints -> Some module/address receives the tokens. (Where the receiving module might be the minter too)

Reply:
>> the idea was to not send and have a functionality and event to mint to an account.
> I don't follow this.

@alexanderbez , currently only module mints to itself (minting is only happening to a module). Minting to an account is done by minting to a module and sending coins from a module to an account. So we can't express in one event minting to an account.

Reply:
For minting to an account: shall we use the same event with an extra attribute (as done in this draft) or it's better to create a new event?

@fdymylja  - will we need to update the Rosetta module?

Reply:
@robert-zaremba, if the current events are mantained as they are then there's no need.

The current implementation uses the mint event to signal a coinbase
Then the addCoins function creates a CoinReceived event. This combo allows rosetta to be aware of balance changes.

Reply:
@fdymylja  - my question was rather related to the changes proposed here: keeping same event with one extra argument. I guess, currently Rosetta assumes that all minting is done to a module only, right?

Reply:
> I guess, currently Rosetta assumes that all minting is done to a module only, right?
Nope, when there's a mint, a `coinbase` event is produced, which identifies how much the supply was inflated, and who is the minter. This event does not make rosetta add coins to the balance of the minter.
What adds coins to the balance of an address is the event produced by `addCoins` bank keeper function which is called in the `Mint` function alongside `SendCoins` etc...

Reply:
I think for consistency we should stick with the current events for now and `MintCoinsToAddress` would then emit two events - one for minting, one for sending. Does that seem workable @robert-zaremba ?

Reply:
OK, so let's do this:
1. We don't create a new event nor extend the current one.
2. All mint events will not have recipient.
3. For Rosetta, to account a new balance, we will use `CoinReceivedEvent` -- triggered in `addCoins`